yewprint-doc/src/icon/mod.rs

```diff
@@ -2,39 +2,198 @@ mod example;
+        .collect::<Vec<_>>()
+});
+
+fn get_icon_from_name(name: &str) -> IconName {
+    let mut res = IconName::Blank;
+
+    for (icon_name, icon) in ICON_LIST.iter() {
+        if name.to_lowercase() == *icon_name.to_lowercase() {
+            res = *icon;
+        }
+    }
+
+    res
+}
 
 impl Component for IconDoc {
-    type Message = ();
+    type Message = IconDocMsg;
     type Properties = ();
 
-    fn create(_ctx: &Context<Self>) -> Self {
-        IconDoc
+    fn create(ctx: &Context<Self>) -> Self {
+        IconDoc {
+            callback: ctx.link().callback(|x| IconDocMsg::Example(x)),
+            state: ExampleProps {
+                icon_name: IconName::Print,
+                intent: None,
+                icon_size: 16,
+            },
+            search_icon: Default::default(),
+        }
     }
 
-    fn update(&mut self, _ctx: &Context<Self>, _msg: Self::Message) -> bool {
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            IconDocMsg::Example(x) => self.state = x,
+            IconDocMsg::SearchIcon(x) => self.search_icon = x,
+        }
         true
     }
 
-    fn view(&self, _ctx: &Context<Self>) -> Html {
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        let example_props = self.state.clone();
         let source = crate::include_raw_html!(
             concat!(env!("OUT_DIR"), "/", file!(), ".html"),
             "bp3-code-block"
         );
 
+        let icon_list = ICON_LIST
+            .iter()
+            .filter_map(|(name, icon)| {
+                name.to_lowercase()
+                    .contains(&self.search_icon.to_lowercase())
+                    .then_some(html! {
+                        <div class={classes!("docs-icon-list-item")}>
+                            <Icon
+                                icon={*icon}
+                                icon_size=20
+                            />
+                            <Text>{name}</Text>
+                        </div>
+                    })
+            })
+            .collect::<Html>();
+
         html! {
             <div>
                 <H1 class={classes!("docs-title")}>{"Icon"}</H1>
                 <SourceCodeUrl />
-                <ExampleContainer source={source}>
-                    <Example />
+                <ExampleContainer
+                    source={source}
+                    props={Some(html! {
+                        <IconProps
+                            callback={self.callback.clone()}
+                            example_props={example_props.clone()}
+                        />
+                    })}
+                >
+                        <Example ..example_props />
                 </ExampleContainer>
+                <div class={classes!("docs-icon-search")}>
+                    <InputGroup
+                        large={true}
+                        fill={true}
+                        round={true}
```

Comment:
You don't need the `{}` for literals
```suggestion
                        large=true
                        fill=true
                        round=true
```